src/main/java/de/tum/in/www1/artemis/service/connectors/localci/LocalCIContainerService.java

```diff
@@ -53,6 +57,8 @@ public class LocalCIContainerService {
 
     private final HostConfig hostConfig;
 
+    private List<BuildLogEntry> buildLogEntries = new ArrayList<>();
```

Comment:
This will still cause the same issue as observed by @sarpsahinalp in case multiple exercises have build jobs running at the same time (#7595):

1. Build job $A$ starts.
2. Build job $A$ writes log messages $l_0$ to $l_n$.
3. Build job $B$ starts and clears the log messages.
4. Build job $B$ writes log messages $k_0, …$.
5. Build job $A$ finishes.
6. The entries list now contains values $l_{n+1}, …$ and $k_{0}, …$ in some randomly mixed order.

Possible solutions:
- make the service properly stateless by making the `List<BuildLogEntries` a local variable of `executeDockerCommand` and return it
- in case you can _uniquely_ identify individual container commands: Find a map structure that allows concurrent access and put buid log entry lists there under the key of the unique id.
  - Drawback: you have to be careful to clean up the map in _all_ exception/error conditions after the build logs have been retrieved. Otherwise the map will be growing over time filling the RAM.